test/e2e/framework/util.go

```diff
@@ -279,35 +278,28 @@ func GetFreePort(t *testing.T) (string, error) {
 
 type ArtifactFunc func(*testing.T, func() (runtime.Object, error))
 
-// CreateClusterAndWait creates a new Cluster resource with the desired name on a given server and waits for it to be ready.
-func CreateClusterAndWait(t *testing.T, ctx context.Context, artifacts ArtifactFunc, kcpClient kcpclientset.Interface, pcluster RunningServer) (*workloadv1alpha1.WorkloadCluster, error) {
+// CreateReadyCluster creates a new Cluster resource with the desired name on a
+// given server and sets it to a ready state.
+func CreateReadyCluster(t *testing.T, ctx context.Context, artifacts ArtifactFunc, kcpClient kcpclientset.Interface, pcluster RunningServer) (*workloadv1alpha1.WorkloadCluster, error) {
 	config, err := pcluster.RawConfig()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get server config: %w", err)
 	}
-	pclusterConfig := clientcmdapi.Config{
-		Clusters: map[string]*clientcmdapi.Cluster{
-			"pcluster": config.Clusters["system:admin"],
-		},
-		Contexts: map[string]*clientcmdapi.Context{
-			"pcluster": {
-				Cluster:  "pcluster",
-				AuthInfo: "admin",
-			},
-		},
-		AuthInfos: map[string]*clientcmdapi.AuthInfo{
-			"admin": config.AuthInfos["admin"],
-		},
-		CurrentContext: "pcluster",
-	}
 
-	bs, err := clientcmd.Write(pclusterConfig)
+	// Assume the supplied configuration does not need to be modified for use.
+	bs, err := clientcmd.Write(config)
 	if err != nil {
 		return nil, fmt.Errorf("failed to serialize server config: %w", err)
 	}
 
+	// The name of the pcluster could be the name of a logical cluster
+	// which is of the form `org:workspace`. Since `:` isn't allowed
+	// in resource names, replacing it with '.' results in a name safe
+	// for use as a resource name.
+	safeClusterName := strings.Replace(pcluster.Name(), ":", ".", 1)
```

Comment:
It's not allowed in CRs afaik, for reasons I never understood.

Reply:
`.` isn't allowed or `:`